packages/ruleset/src/functions/composed-schema-restrictions.js

```diff
@@ -1,123 +1,44 @@
-  return checkSchemaList(schema, subSchemas, fieldName, path);
-}
-
-/**
- * This function checks "schema" and its oneOf/anyOf list "subSchemas" to verify that
- * it complies with the generator restrictions around the use of oneOf/anyOf.
- * @param {*} schema the main schema to check
- * @param {*} subSchemas the oneOf or anyOf list belonging to "schema"
- * @param {*} fieldName the name of the field containing "subSchemas" within "schema" ('oneOf' or 'anyOf')
- * @param {*} path the array of path segments indicating the location of "schema" within the API definition
- * @returns an array containing the violations found or [] if no violations
- */
-function checkSchemaList(schema, subSchemas, fieldName, path) {
   const errors = [];
 
-  schema = mergeAllOfSchemaProperties(schema);
-
-  // "targetProperties" will initially contain the properties defined in the main schema,
-  // then we'll add each property from the subSchemas to it to simulate the SDK generator's
-  // handling of the oneOf/anyOf list (hybrid hierarchy mode).
-  const targetProperties = isObject(schema.properties)
-    ? copyObject(schema.properties)
-    : {};
-
-  for (let i = 0; i < subSchemas.length; i++) {
-    const s = mergeAllOfSchemaProperties(subSchemas[i]);
-
-    // "s" should be an object schema. If not, return an error.
-    if (!isObjectSchema(s)) {
-      errors.push({
-        message: `A schema within an object schema's oneOf/anyOf list must be an object schema`,
-        path: [...path, fieldName, i.toString()]
-      });
-    } else {
-      // "s" is an object schema as expected, so let's check each of its properties.
-      const properties = isObject(s.properties) ? s.properties : {};
-      for (const [name, prop] of Object.entries(properties)) {
-        const targetProp = targetProperties[name];
-        if (targetProp) {
-          // If "targetProperties" contains a like-named property, make sure the types are compatible.
-          // We can only guess at the paths associated with target and sub-schema properties
-          // due to the allOf processing that needs to be performed on each schema.
-          const propsAreCompatible = propsHaveCompatibleTypes(
-            targetProp,
-            [...path, 'properties', name],
-            prop,
-            [...path, fieldName, i, 'properties', name]
-          );
-          if (!propsAreCompatible) {
-            const error = {
-              message: `Duplicate property with incompatible type defined in schema within a oneOf/anyOf list: ${name}`,
-              path: [...path, fieldName, i.toString()]
-            };
-            errors.push(error);
-          }
-        } else {
-          // Property doesn't exist in main schema, so just add it.
-          targetProperties[name] = prop;
-        }
+  // Only object schemas have properties
+  if (isObjectSchema(schema)) {
+    // Collects all composed property schemas indexed by the name of the property they define
```

Comment:
Would be easier to read (for me, at least) if this commented elaborated slightly more on the data structure returned by `getPropertySchemasByName`. Something to let the reader know the map values are lists of schemas all defining properties of common names.